src/Controller/CommitmentController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Dto\CommitmentIn;
+use App\Dto\CommitmentOut;
+use App\Entity\Commitment;
+use App\Entity\Order;
+use App\Repository\OrderRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityNotFoundException;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+class CommitmentController
+{
+    private OrderRepository $orderRepository;
+    private DenormalizerInterface $denormalizer;
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(
+        OrderRepository $orderRepository,
+        DenormalizerInterface $denormalizer,
+        EntityManagerInterface $entityManager
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->denormalizer = $denormalizer;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @Route(
+     *     "/commitment",
```

Comment:
```suggestion
     *     "/commitments",
```

Reply:
Da wird nur ein Commitment erstellt, also ist die Einzahl eigentlich korrekt

Reply:
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/